### ISSUE TITLE
Reduce reporting to a singe check or add option to disable WIP

### ISSUE THREAD

**Issue by jsumners** [amannn/action-semantic-pull-request#27]
I have a simple test repo where I am exploring usage of this action. I created a branch on this repo, added some text to a markdown file, and pushed up a commit with an invalid message. The resulting PR reports a status like:

![image](https://user-images.githubusercontent.com/321201/93352293-adf28380-f808-11ea-9e3c-36f909dbe60b.png)

I have a `.github/workflows/pr-commit-lint.yml` file that looks like:

```yaml
name: 'Lint Commit Messages'
on:
  pull_request_target:
    types:
      - opened
      - edited
      - synchronize

jobs:
  main:
    runs-on: ubuntu-latest
    steps:
      - uses: amannn/action-semantic-pull-request@v1.2.0
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Is this expected behavior?

**Comment by amannn** [comment]
Thanks for your report! I agree, it's definitely a bit confusing that there are two status checks. Especially when the one is failing and the other one says ready for merge.

I've just merged a PR which improves the situation for failed checks a bit: https://github.com/amannn/action-semantic-pull-request/pull/30.

The reason why there are two checks is that one is for the validation itself and the other one creates a pending check that doesn't complete for the special `[WIP]` syntax. The updated readme explains why this is supported and that you might not need it if your repository supports draft pull requests.

What I'd definitely be open for, is adding an option that disables the WIP check. In this case you'll only have one status check. It would essentially make the block you mentioned above conditional.

Would you be interested in contributing this option?

**Comment by jsumners** [comment]
From #30:
> If you click it, it should contain more details about what went wrong. In this case I'm linking to this repository, where someone who might not be familiar with the validation can get more information.

That is very confusing. The typical behavior of that "Details" link is to navigate to the Action report to see the output of the action. If the link is going to continue linking elsewhere, it should be to an anchor that explains the behavior and why.



> The reason why there are two checks is that one is for the validation itself and the other one creates a pending check that doesn't complete for the special `[WIP]` syntax. The updated readme explains why this is supported and that you might not need it if your repository supports draft pull requests.

The readme update doesn't explain that two checks will be generated by the single action. That would at least help.

I understand now why you are generating the two checks. But it is odd, unexpected, behavior. I know you addressed it in #21, and I agree with the reasoning, but I think it might be clearer if the action generates a single check that is either "pending" or "failed" in the case of `[WIP] ` being present; actually, if "skipped" is possible then I think that would be ideal. Once the pseudo-tag, or the draft status, is removed, then the action should complete its course and report the true status.



> Would you be interested in contributing this option?

I would, but unfortunately I am unable to do so at this time.

**Comment by amannn** [comment]
> might be clearer if the action generates a single check

Yes, I also think that would be the proper solution – everything else is a bit a workaround. When I initially built the action, I couldn't figure this out, but maybe I have overlooked something or the situation might have changed since then.

> I would, but unfortunately I am unable to do so at this time.

I also don't have time to look into it currently, but let's leave this issue open – maybe someone else is interested as well 🙂 .

**Comment by amannn** [comment]
I've made the WIP feature which causes the second check now opt-in. Therefore the common case will only have a single check now.

Released in [3.0.0](https://github.com/amannn/action-semantic-pull-request/releases/tag/v3.0.0)

**Comment by jsumners** [comment]
🎉
